### ISSUE TITLE
-DCMAKE_BUILD_TYPE=Debug: No rule to make target 'ext/makefiles/quazip/project/lib/libquazip5.so', needed by 'plugins/hifiSdl2/libhifiSdl2.so'.

### ISSUE THREAD

**Issue by ChristophHaag** [highfidelity/hifi#12938]
Current Archlinux, cmake 3.10.3.
I try to compile with
```
QT_CMAKE_PREFIX_PATH=/usr/lib/cmake cmake -DCMAKE_BUILD_TYPE=Debug ../hifi
make -j11 interface
```
and get this error:
```
[ 62%] Building CXX object plugins/hifiSdl2/CMakeFiles/hifiSdl2.dir/hifiSdl2_autogen/mocs_compilation.cpp.o
make[3]: *** No rule to make target 'ext/makefiles/quazip/project/lib/libquazip5.so', needed by 'plugins/hifiSdl2/libhifiSdl2.so'.  Stop.
```

I have not found out why this happens. Complete console output:
[compileoutput.txt](https://github.com/highfidelity/hifi/files/1926026/compileoutput.txt)

I can only say for sure that the error got introduced somewhere between
RELEASE-7995 - builds fine
RELEASE-8108 - build error

I git bisected it twice and twice my result was this merge commit: 621e6a62d6155b3629f6017a92dd3a3df8e3c831.
Only it does not make any sense because this commit touches no cmake files, and contains nothing about libhifiSdl2 or libquazip.

Currently I'm out of ideas but if anyone has an idea, I'll be happy to test anything.

**Comment by ChristophHaag** [comment]
Updated the title because I noticed that the library that was built was called ext/makefiles/quazip/project/lib/libquazip5**d**.so

The (now removed) glew library did something similar and declared two different library names depending on the build type: https://github.com/highfidelity/hifi/blob/RELEASE-7607/cmake/externals/glew/CMakeLists.txt#L33-L34
And depending on the build type other build files probably depended on the different library names.

**Comment by sethalves** [comment]
I've just experienced this also (ubuntu 16.04)

**Comment by sethalves** [comment]
does this work?
```
diff --git a/cmake/externals/quazip/CMakeLists.txt b/cmake/externals/quazip/CMakeLists.txt
index f2690e0..7bf6f05 100644
--- a/cmake/externals/quazip/CMakeLists.txt
+++ b/cmake/externals/quazip/CMakeLists.txt
@@ -41,6 +41,9 @@ if (APPLE)
 elseif (WIN32)
   set(${EXTERNAL_NAME_UPPER}_LIBRARY_RELEASE ${INSTALL_DIR}/lib/quazip5.lib CACHE FILEPATH "Location of QuaZip release library")
   set(${EXTERNAL_NAME_UPPER}_LIBRARY_DEBUG ${INSTALL_DIR}/lib/quazip5d.lib CACHE FILEPATH "Location of QuaZip release library")
+elseif (CMAKE_SYSTEM_NAME MATCHES "Linux")
+  set(${EXTERNAL_NAME_UPPER}_LIBRARY_RELEASE ${INSTALL_DIR}/lib/libquazip5.so CACHE FILEPATH "Location of QuaZip release library")
+  set(${EXTERNAL_NAME_UPPER}_LIBRARY_DEBUG ${INSTALL_DIR}/lib/libquazip5d.so CACHE FILEPATH "Location of QuaZip release library")
 else ()
   set(${EXTERNAL_NAME_UPPER}_LIBRARY_RELEASE ${INSTALL_DIR}/lib/libquazip5.so CACHE FILEPATH "Location of QuaZip release library")
   set(${EXTERNAL_NAME_UPPER}_LIBRARY_DEBUG ${INSTALL_DIR}/lib/libquazip5.so CACHE FILEPATH "Location of QuaZip release library")
```

**Comment by ChristophHaag** [comment]
It does.

**Comment by AndrewMeadows** [comment]
I'm closing this for **fixed** in #13050.
